src/main/java/io/codelex/flightplanner/flights/inmemory/FlightInMemoryRepository.java

```diff
@@ -0,0 +1,67 @@
+package io.codelex.flightplanner.flights.inmemory;
+
+import io.codelex.flightplanner.flights.FlightRepository;
+import io.codelex.flightplanner.flights.domain.Airport;
+import io.codelex.flightplanner.flights.domain.Flight;
+import io.codelex.flightplanner.flights.domain.SearchFlightRequest;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+
+@Repository
+@ConditionalOnProperty(prefix = "flightplanner", name = "appmode", havingValue = "inmemory")
+public class FlightInMemoryRepository implements FlightRepository {
+
+    private List<Flight> flightList = new CopyOnWriteArrayList<>();
+
+    public List<Flight> searchFlights(SearchFlightRequest request) {
+        List<Flight> list = flightList.stream().filter(flight ->
+                flight.getFrom().getAirport().equals(request.getFrom()) &&
+                        flight.getTo().getAirport().equals(request.getTo()) &&
+                        flight.getDepartureDate().equals(request.getDepartureDate())).toList();
+        return list;
+    }
+
+    public Airport searchAirports(String search) {
+        String lowerCaseSearch = search.trim().toLowerCase();
+        Flight ifFoundInGetToAirport = flightList.stream().filter(flight ->
```

Comment:
Varbūt ir vērts iekš Airport uztaisīt metodi - `boolean airportInfoContains(String text)` kura pārbaudītu vai konkrētā Airport dati satur padoto tekstu.
Tad šiet paliktu tikai :
`flightList.stream().filter(flight -> flight.getTo().airportInfoContains(lowerCaseSearch)).findFirst().orElse(null);`
Kad šito izdari - vari uzrakstīt man ziņu, jo ir variants kā tikt vaļā no tā, ka ir divreiz search jātaisa - gan ar from gan ar to lidostām.